File: BoolToggleLightSwitch/BoolToggleLightSwitch-ObjC/ViewController.m
//
//  ViewController.m
//  BoolToggleLightSwitch
//
//  Created by Ibrahim Hassan on 16/01/19.
//  Copyright © 2019 None. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property BOOL isLightOn;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (IBAction)lightFlipped:(UIButton *)sender {
    self.isLightOn = [self toggle:self.isLightOn];
    if (self.isLightOn == false) {
        [sender setTitleColor:UIColor.whiteColor forState:UIControlStateNormal];
        self.view.backgroundColor = UIColor.blackColor;
    } else {
        [sender setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
        self.view.backgroundColor = UIColor.whiteColor;
    }
}

-(BOOL)toggle:(BOOL) value {
    return !value;
}

@end

File: BoolToggleLightSwitch/BoolToggleLightSwitch-Swift/ViewController.swift
//
//  ViewController.swift
//  BoolToggleLightSwitch-Swift
//
//  Created by Ibrahim Hassan on 16/01/19.
//  Copyright © 2019 None. All rights reserved.
//

import UIKit

// http://swiftify.me/tg7fb6
//  Converted to Swift 4 by Swiftify v4.2.20418 - https://objectivec2swift.com/
class SwiftViewController : UIViewController {
    var isLightOn = false
    
    @IBAction func lightFlipped(_ sender: UIButton) {
        isLightOn.toggleValue()
//        isLightOn.toggle()
//        isLightOn = toggle(isLightOn)
        // After Swift 4.2
        isLightOn.toggle()
        if isLightOn == false {
            sender.setTitleColor(UIColor.white, for: .normal)
            view.backgroundColor = UIColor.black
        } else {
            sender.setTitleColor(UIColor.black, for: .normal)
            view.backgroundColor = UIColor.white
        }
    }
    
    func toggle(_ value: Bool) -> Bool {
        return !value
    }
}

// Toggle before Swift 4.2
extension Bool {
    
    mutating func toggleValue() {
        self = !self
    }
    
}
